xonsh/prompt/gitstatus.py

```diff
@@ -110,20 +119,28 @@ def gitstatus():
     stashed = _get_stash(gitdir)
     operations = _gitoperation(gitdir)
 
-    return (branch, num_ahead, num_behind,
+    return GitStatus(branch, num_ahead, num_behind,
             untracked, changed, conflicts, staged, stashed,
             operations)
 
 
 def gitstatus_prompt():
     """Return str `BRANCH|OPERATOR|numbers`"""
     try:
-        (branch, num_ahead, num_behind,
-         untracked, changed, conflicts, staged, stashed,
-         operations) = gitstatus()
+        status = gitstatus()
     except subprocess.SubprocessError:
         return None
 
+    branch = status.branch
+    num_ahead = status.num_ahead
+    num_behind = status.num_behind
+    untracked = status.untracked
+    changed = status.changed
+    conflicts = status.conflicts
+    staged = status.staged
+    stashed = status.stashed
+    operations = status.operations
```

Comment:
I usually use namedtuples as compact data structs. The point is that you kinda forget the order of it.